apps/full-site-editing/full-site-editing-plugin/full-site-editing-plugin.php

```diff
@@ -40,6 +40,42 @@
 // Always include these helper files for dotcom FSE.
 require_once __DIR__ . '/dotcom-fse/helpers.php';
 
+/**
+ * Load Core Site Editor.
+ */
+function load_core_site_editor() {
+	$options = get_option( 'gutenberg-experiments' );
+	// Initialize options array if needed.
+	if ( ! $options ) {
+		$options = array();
+	}
+	// Check blog sticker for access to Site Editor.
+	if ( ! has_blog_sticker( 'core_site_editor_enabled' ) ) {
+		// Check if experiment is enabled: disable if needed.
+		if ( gutenberg_is_experiment_enabled( 'gutenberg-full-site-editing' ) ) {
+			$options['gutenberg-full-site-editing']      = '0';
+			$options['gutenberg-full-site-editing-demo'] = '0';
+			update_option( 'gutenberg-experiments', $options );
+		}
+		return;
+	}
+	// Check if experiment is disabled: enable if needed.
+	if ( ! gutenberg_is_experiment_enabled( 'gutenberg-full-site-editing' ) ) {
+		$options['gutenberg-full-site-editing']      = '1';
+		$options['gutenberg-full-site-editing-demo'] = '1';
+		update_option( 'gutenberg-experiments', $options );
+	}
+	add_menu_page(
+		__( 'Site Editor (beta)', 'gutenberg' ),
+		__( 'Site Editor (beta)', 'gutenberg' ),
+		'edit_theme_options',
+		'gutenberg-edit-site',
+		'gutenberg_edit_site_page',
```

Comment:
This callback does seem to be firing, as the blank page loaded does contain the `edit-site` div.

I am thinking the editor is blank because `gutenberg_edit_site_init` function is never firing.

It looks like this init function is only set to the `admin_enqueue_scripts` action:
https://github.com/WordPress/gutenberg/blob/d026d5b5c189e85bf1bb83321a74494c664e0159/lib/edit-site-page.php#L157

And returns early if it doesn't receive the `gutenberg_page_gutenberg-edit-site` hook:
https://github.com/WordPress/gutenberg/blob/d026d5b5c189e85bf1bb83321a74494c664e0159/lib/edit-site-page.php#L37-L39

🤔  -  I wonder how we can get this init to fire from here.  🤔

Reply:
I have tried both changing the slug we use on the `add_menu_page`, as well as recreating the standard menu hierarchy.   But no luck with either of those.

Reply:
> It looks like this init function is only set to the admin_enqueue_scripts

Interesting... That definitely seems tricky. Do you know where the `gutenberg_page_gutenberg-edit-site` hook comes from? You could try adding some logs to the gutenberg plugin files and then looking at them on your sandbox if that helps :)

Reply:
It turns out that re-creating the original page hierarchy (gutenberg page -> site editor subpage) does generate the proper slug and gets the init function to run past its early return statement.  However, this is still resulting in a blank page in the end.

Im still digging but I'm not quite sure if this is a result of dotcom using an older version of gutenberg, or if it has to do with how we incorporate the gutenberg plugin (or both?).

>  Do you know where the gutenberg_page_gutenberg-edit-site hook comes from?

`gutenberg_page_gutenberg-edit-site` comes from the plugin page hierarchy when a selection is made.  It ends up being `$pageSlug_page_$subpageSlug`.  So with the core implementation `gutenberg-edit-site` is the subpage set on the `gutenberg` page, resulting in that full slug for the hook when someone clicks the 'Site Editor (beta)' button.  Putting the edit site at the page level results in a slug similar to `front-page_page_gutenberg-edit-site`.